packages/dx-grid-core/src/plugins/custom-grouping/computeds.test.js

```diff
@@ -0,0 +1,247 @@
+import {
+  customGroupedRows,
+  customGroupingRowIdGetter,
+} from './computeds';
+import {
+  GRID_GROUP_TYPE,
+  GRID_GROUP_CHECK,
+  GRID_GROUP_LEVEL_KEY,
+} from '../local-grouping/constants';
+
+describe('CustomGrouping Plugin computeds', () => {
+  describe('#customGroupedRows', () => {
+    it('should process hierarchical data by one column', () => {
+      const hierarchicalSource = [{
+        key: 1,
+        items: [
+          { a: 1, b: 1 },
+          { a: 1, b: 2 },
+        ],
+      }, {
+        key: 2,
+        items: [
+          { a: 2, b: 1 },
+          { a: 2, b: 2 },
+        ],
+      }];
+      const getHierarchicalChildGroups = groups => groups
+        .map(group => ({ key: String(group.key), value: group.key, nestedData: group.items }));
```

Comment:
Can we extract duplicated code? I mean something like this:
```js
const createGetChildGroupsMock = (groups) => {
  const getHierarchicalChildGroups = groups => groups
        .map(group => ({ key: String(group.key), value: group.key, nestedData: group.items }));
  return jest.fn(getHierarchicalChildGroups);
};
```

Reply:
I don't like decreasing tests readability as it will not reduce loc count dramatically.